### PR TITLE
Fix CLI and some refactoring

### PR DESCRIPTION
**Pull request by eugenkleinschmidt**
- fix CLI commands `set-ports` and `toggle-ports`
- refactor _execute_try -> remove lambda. Use stronger type and better error message

### CODE REVIEW

**Comment by eugenkleinschmidt** [comment]
@stephrdev  What about removing the check and raise here? It should be already ok with the type

```
class RequestFrame:
    def __init__(self, command: CommandCodes, address: int = 0, data: int = 0) -> None:
        if command not in CommandCodes:
            raise RelayCardError(f"Wrong command{command}. Not found in {CommandCodes}")
        self.command = command
```

File: src/relaycard/card.py [hunk 1]
```diff
@@ -135,37 +138,26 @@ def setup(self) -> bool:
         return self.is_initialized
 
     def get_ports(self, address: int) -> RelayState:
-        response = self._execute_retry(CommandCodes.GETPORT, lambda r: r.command == ResponseCodes.GETPORT, address)
+        response = self._execute_retry(ComCodes.GETPORT, address)
         return RelayState(response.data)
 
     def get_port(self, address: int, port: int) -> bool:
-        if not (0 <= port <= 7):
```
Comment (stephrdev) [inline src/relaycard/card.py:142]: Why don't we validate the given port number anymore?

**Comment by stephrdev** [comment]
> @stephrdev What about removing the check and raise here? It should be already ok with the type

Agreed.

**Comment by stephrdev** [comment]
Some stuff I noticed:

- [ ] Card is initialized even if no card-command like get-ports/set-ports is used (for example running without any command at all)
- [ ] Default serial port is defined in two places. Couldn't we remove the default from the Card class and only have it in the cli?

File: src/relaycard/card.py (hunk 1 above)
Comment (eugenkleinschmidt) [reply src/relaycard/card.py:142]: It is done by `RelayState.get_port`. So it was duplicated

**Comment by eugenkleinschmidt** [comment]
> > @stephrdev What about removing the check and raise here? It should be already ok with the type
> 
> Agreed.

Done

**Comment by eugenkleinschmidt** [comment]
> Some stuff I noticed:
> 
>     * [x]  Card is initialized even if no card-command like get-ports/set-ports is used (for example running without any command at all)
> 
>     * [x]  Default serial port is defined in two places. Couldn't we remove the default from the Card class and only have it in the cli?

I think the "action" commands could be made into a exclusic group and put some args as required

**Comment by eugenkleinschmidt** [comment]
Anything else missing for this PR?

**Comment by stephrdev** [comment]
No, i’m on vacation and had no chance to check an merge. Will do asap

**Pull request merged**